api/compaction_manager.cc

```diff
@@ -74,14 +74,15 @@ void set_compaction_manager(http_context& ctx, routes& r) {
 
     cm::get_pending_tasks_by_table.set(r, [&ctx] (std::unique_ptr<request> req) {
         return ctx.db.map_reduce0([&ctx](database& db) {
-            std::unordered_map<std::pair<sstring, sstring>, uint64_t, utils::tuple_hash> tasks;
+            return do_with(std::unordered_map<std::pair<sstring, sstring>, uint64_t, utils::tuple_hash>(), [&ctx, &db](std::unordered_map<std::pair<sstring, sstring>, uint64_t, utils::tuple_hash>& tasks) {
             return do_for_each(db.get_column_families(), [&tasks](const std::pair<utils::UUID, seastar::lw_shared_ptr<table>>& i) {
                 table& cf = *i.second.get();
                 tasks[std::make_pair(cf.schema()->ks_name(), cf.schema()->cf_name())] = cf.get_compaction_strategy().estimated_pending_compactions(cf);
                 return make_ready_future<>();
             }).then([&tasks] {
                 return tasks;
```

Comment:
I hope this copies tasks *before* returning. I think it does, i.e., it returns a future<tasks>, not future<tasks&>, otherwise I hope this code wouldn't compile. Please do make sure to test that this code actually works in a real test that exercises it.

Reply:
Maybe it's even one of these rare situations where "return std::move(tasks)" is exactly the right thing to do? It returns a new map object, but moves its actual content, relatively efficiently, from the old map object, and will save a lot of deallocations and allocations.

Reply:
It's hard to tell with testing, it worked for me when I tested it even when it was wrong, even with the test that found the issue, it's not always reproducible.

Reply:
I'll add an std::move, it's an easy change